### ISSUE TITLE
Question : Why the _backward i not used from the SamplerQNN  

### ISSUE THREAD

**Issue by invochex** [qiskit-community/qiskit-machine-learning#895]
Hello,

I am currently exploring Deep Learning and Quantum Computing for my research, and I am using qiskit_machine_learning to implement a Variational Quantum Classifier (VQC) for binary classification. Below is my model:

```
from qiskit_machine_learning.algorithms import VQC
import time

# Creating a VQC instance
vqc = VQC(
    sampler=sampler,
    optimizer= classical_optimizer(),
    feature_map= feature_map(),
    ansatz= variational_circuit(),
    callback=callback_graph,
)

# Model Training
objective_func_vals = []
start = time.time()
vqc.fit(train_features, train_labels)
elapsed = time.time() - start

print(f"Training time: {round(elapsed)} seconds")

# Evaluation performance
score = vqc.score(train_features, train_labels)
print(f"Accuracy: {score}")
```

Here is the sampler:

```
from qiskit.primitives import StatevectorSampler as Sampler

sampler = Sampler()
```

Here is the optimizer using COBYLA:

```
def classical_optimizer():
    from qiskit_algorithms.optimizers import COBYLA
    
    cls_opt = COBYLA(maxiter=100, disp=True, tol=1e-6)
    return cls_opt

```
I need some clarification regarding the backpropagation process in Quantum Neural Networks (QNNs), particularly in Sampler_QNN.

I noticed that the measurement of my quantum circuit is performed in the _forward function of Sampler_QNN, which is used by default in VQC.

However, I expected _backward to be used at some point for gradient optimization.

I inserted a print statement in _backward, but it was never executed, whereas _forward was called 101 times.

# Questions:

Why is _backward never used in Sampler_QNN?

I assumed that _backward is responsible for optimizing gradients. If not, how does optimization work in a QNN like Sampler_QNN?

Since COBYLA is a classical optimizer, does that mean optimization is performed in a classical way?

What does backpropagation correspond to in quantum neural networks?

Does the weight update happen every time _forward is called, or is there another mechanism?

I am struggling to fully grasp the underlying principle and would greatly appreciate any insights!

Thank you in advance!

**Comment by OkuyanBoga** [comment]
> Why is _backward never used in Sampler_QNN?
> I assumed that _backward is responsible for optimizing gradients. If not, how does optimization work in a QNN like Sampler_QNN?

You are using COBYLA, which is a gradient-free optimiser. Please see the [description](https://docs.scipy.org/doc/scipy/reference/optimize.minimize-cobyla.html).

>  Does the weight update happen every time _forward is called, or is there another mechanism?
Weights are updated when you call .fit(), never in forward, unless it is the forward-forward algorithm 😄 

> Since COBYLA is a classical optimizer, does that mean optimization is performed in a classical way?

Most of the optimizers in the variational algorithms are classical, please check 'variational quantum algorithms/classifiers' to understand how parameter optimization is done.
